src/Microsoft.DotNet.Build.Tasks/GetPackageVersion.cs

```diff
@@ -34,7 +35,13 @@ public override bool Execute()
                 select el).FirstOrDefault();
 
             Debug.Assert(embeddedVerNode != null, "embeddedVerNode shouldn't be null");
-            PackageVersion = string.Format("{0}-{1}", embeddedVerNode.Value, BuildNumber);
+
+            // if this is a prerelease version then append the build number to the end
+            Regex preReleaseMatch = new Regex("-[0-9A-Za-z]+$");
```

Comment:
I assume you don't actually need to "0-9" as it should only be chars in this spot correct? While I think this solves our issue for now I wonder if we need to explicitly pass in the "prerelease" string or not instead of having that in the nuspec.